cms/templates/admin/cms/page/history/recover_header.html

```diff
@@ -1,22 +1,26 @@
 {% extends "admin/cms/page/history/revision_header.html" %}
-{% load i18n cms_tags %}
+{% load i18n cms_tags admin_urls %}
 
 {% block extrahead %}
-{{ block.super }}
-<script src="{% cms_admin_url "jsi18n" %}" type="text/javascript"></script>
-{{ media }}
+    {{ block.super }}
+    <script src="{% cms_admin_url "jsi18n" %}" type="text/javascript"></script>
+    {{ media }}
 {% endblock %}
 
 {% block breadcrumbs %}
-<div class="breadcrumbs">
-    <a href="../../../../">{% trans "Home" %}</a> &rsaquo;
-    <a href="../../../">{{ app_label|capfirst|escape }}</a> &rsaquo;
-    <a href="../../">{{ opts.verbose_name_plural|capfirst }}</a> &rsaquo;
-    <a href="../">{% blocktrans with opts.verbose_name as verbose_name %}Recover deleted {{ verbose_name }}{% endblocktrans %}</a> &rsaquo;
-    {{ title }}
-</div>
+    <div class="breadcrumbs">
+        <a href="{% url 'admin:index' %}">{% trans "Home" %}</a> &rsaquo;
+        <a href="{% url 'admin:app_list' app_label=opts.app_label %}">{{ opts.app_config.verbose_name }}</a> &rsaquo;
+        <a href="{% url opts|admin_urlname:'changelist' %}">{{ opts.verbose_name_plural|capfirst }}</a> &rsaquo;
+        <a href="../">
```

Comment:
@FinalAngel where this is used? if you have the current object, something like ``{% url opts|admin_urlname:'change' object.pk %}`` should work